engine/apps/api/views/alert_group.py

```diff
@@ -337,6 +331,16 @@ def get_queryset(self, ignore_filtering_by_available_teams=False):
         if not ignore_filtering_by_available_teams:
             alert_receive_channels_qs = alert_receive_channels_qs.filter(*self.available_teams_lookup_args)
 
+        # Filter by team(s). Since we really filter teams from integrations, this is not an AlertGroup model filter.
+        # This is based on the common.api_helpers.ByTeamModelFieldFilterMixin implementation
```

Comment:
This is the main change. Removing team from filter above will not add the team condition in the alert groups query. Instead prefer to restrict the team when filtering alert receive channels, which we are already using to filter alert groups per org.